cmd/gdsutil/main.go

```diff
@@ -1110,6 +1132,41 @@ func findCertificateRequest(vaspID string) (cr *models.CertificateRequest, err e
 	return nil, nil
 }
 
+func generateTokenKey(c *cli.Context) (err error) {
+	// Create ksuid and determine outpath
+	var keyid ksuid.KSUID
+	if keyid, err = ksuid.NewRandom(); err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	var out string
+	if out = c.String("out"); out == "" {
+		out = fmt.Sprintf("%s.pem", keyid)
+	}
+
+	// Generate RSA keys using crypto random
+	var key *rsa.PrivateKey
+	if key, err = rsa.GenerateKey(rand.Reader, c.Int("size")); err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	// Open file to PEM encode keys to
+	var f *os.File
+	if f, err = os.OpenFile(out, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600); err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	if err = pem.Encode(f, &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(key),
+	}); err != nil {
+		return cli.NewExitError(err, 1)
+	}
+
+	fmt.Printf("RSA key id: %s -- saved with PEM encoding to %s\n", keyid, out)
```

Comment:
Note that an RSA private key contains the public key; so we can extract the public key from the private key without having to write it separately.